lib/work_q.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2016 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#define WORKQUEUE_THREAD_NAME	"workqueue"
+
+void z_work_q_main(void *work_q_ptr, void *p2, void *p3)
```

Comment:
Just wondering why these 2 functions were kept in lib and not in kernel. This is invoked by the kernel.

Reply:
Never mind got the answer.

Reply:
to anyone else reviewing: `z_work_q_main()` might not run in supervisor mode, and hence shouldn't be in kernel/, since all code in there is assumed to be supervisor-only.